api/src/main/java/org/apache/iceberg/metrics/DefaultCounter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.metrics;
+
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/** A default {@link Counter} implementation that uses an {@link AtomicLong} to count events. */
+public class DefaultCounter implements Counter {
+  private final AtomicLong counter;
```

Comment:
We may want to use a `LongAdder` here. Looks like that's built into java.util.concurrent and can handle contention better.

Reply:
sgtm, since we're already using `LongAdder` in the `DefaultTimer` it also makes sense to use it here